framework/src/packets/ip/v6/srh.rs

```diff
@@ -491,6 +487,53 @@ impl<E: Ipv6Packet> IpPacket for SegmentRouting<E> {
 
 impl<E: Ipv6Packet> Ipv6Packet for SegmentRouting<E> {}
 
+impl<E: Ipv6Packet> SegmentsIndex for SegmentRouting<E> {
+    /// Returns the first segment in the segment list.
+    #[inline]
+    fn first_segment(&self) -> Segment {
+        let segments = self.segments();
+        segments[segments.len() - 1]
+    }
+
+    /// Returns the last segment in the segment list.
+    #[inline]
+    fn last_segment(&self) -> Segment {
+        self.segments()[0]
+    }
+
+    /// Returns the current segment in the segment list.
+    #[inline]
+    fn current_segment(&self) -> Segment {
```

Comment:
Ok I was wrong so turns out with an invalid packet, this can err. See my other comment.